### PR TITLE
[db] use Readiness to check DB started

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

**Review by Liuhaai — APPROVED** [review]

File: blockindex/indexbuilder_test.go [hunk 1]
```diff
@@ -68,16 +68,14 @@ func TestIndexBuilder(t *testing.T) {
...
-			require.NoError(indexer.Stop(ctx))
-			require.NoError(dao.Stop(ctx))
-		}()
-
 		ib := &IndexBuilder{
 			dao:     dao,
 			indexer: indexer,
 		}
-		defer ib.Stop(context.Background())
+		defer func() {
```
Comment (CoderZhi) [inline blockindex/indexbuilder_test.go:75]: why is this change needed in this PR?

Comment (dustinxie) [reply blockindex/indexbuilder_test.go:75]: the original test is not properly written

ib is a wrapper of indexer, `ib.Stop()` calls `ib.indexer.Stop()` inside --> so `indexer.Stop()` is called twice

**Review by CoderZhi — APPROVED** [review]
